Sources/Misc/TimingUtil.swift

```diff
@@ -0,0 +1,168 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  TimingUtil.swift
+//
+//  Created by Nacho Soto on 11/15/22.
+
+import Foundation
+
+internal enum TimingUtil {
+
+    typealias Duration = TimeInterval
+
+}
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+extension TimingUtil {
+
+    /// Measures the time to execute `work` and returns the result and the duration.
+    /// Example:
+    /// ```swift
+    /// let (result, duration) = try await TimingUtil.measure {
+    ///    return try await asyncMethod()
+    /// }
+    /// ```
+    static func measure<Value>(
```

Comment:
I'm not a huge fan of static methods for stuff like this because they complicate DI, i.e: if we wanted to test that some consumer of this method is indeed calling it, static funcs will make that a bit more awkward

Reply:
I'm on the same page, using `static` stuff all over the place is a big smell. 

I wish we had a proper DI framework though, cause this would require wiring this down 3 or 4 layers down.
I'm torn, I'd say YAGNI, but we could inject it later if we need to?

Reply:
> if we wanted to test that some consumer of this method is indeed calling it, static funcs will make that a bit more awkward

I'd say this might not be that critical of a thing to test. If we stopped calling it all we'd lose is a log.

We can also test that we're calling this using `TestLogger`.

Reply:
I'll go ahead and merge this since we can easily change this to be non-static to improve testing (or use `TestLogger`).